# **Live Code 3**
### **HACTIV8 Full Time Data Science Phase 0**

Nama  : Ropiudin

Batch : FTDS OO2-HCK

Live Code : Live Code 3

# **Problem Statement**

Meninjau banyaknya kasus pencurian di kota Austin selama tahun 2016

# **Success Criteria berdasarkan SMART**

S : Mencari informasi jenis tindak kejahatan yang terjadi di kota Austin selama tahun 2016 dan mengetahui banyaknyak kasus pencurian di kota Austin

M : Dapat meninjau banyaknya kasus pencurian di kota Austin selama tahun 2016, sehingga bisa menjadi bahan pertimbangan pihak yang berwenang

A : Berdasarkan penjabaran masalah akan diketahui beberapa informasi yang berkaitan dengan jenis tindak kejahatan yang terjadi di kota Austin, sehingga realistis untuk meninjau banyaknya kasus pencurian di kota Austin selama tahun 2016

R : Sesuai dengan OKR pemerintah

T : Dilakukan pada tahun 2016

# **Working Area**

## **Penjabaran Masalah**

In [1]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

from google.cloud import bigquery
project_id = "ropiudin" 
client = bigquery.Client(project=project_id)

Authenticated


### 1. Tampilkan tabel crime sampai 5 baris pertama

In [3]:
df1 = client.query('''
SELECT *
FROM `bigquery-public-data.austin_crime.crime` LIMIT 5
''').to_dataframe()

df1

unique_key     address census_tract            clearance_date  \
0  20165002898  Austin, TX         None 2016-03-14 12:00:00+00:00   
1  20165004843  Austin, TX         None 2016-03-31 12:00:00+00:00   
2   2016250988  Austin, TX         None 2016-09-16 12:00:00+00:00   
3  20165003839  Austin, TX         None 2016-02-09 12:00:00+00:00   
4   2016321746  Austin, TX         None 2016-03-22 12:00:00+00:00   

  clearance_status council_district_code                     description  \
0      Not cleared                  None  RAPE                             
1      Not cleared                  None  RAPE                             
2      Not cleared                  None  RAPE                             
3      Not cleared                  None  AGG RAPE OF A CHILD              
4      Not cleared                  None  RAPE OF A CHILD                  

  district latitude longitude location location_description primary_type  \
0       UK     None      None     None                  nan         Rape   
1       UK     None      None     None                  nan         Rape   
2       UK     None      None     None                  nan         Rape   
3       UK     None      None     None                  nan         Rape   
4       UK     None      None     None                  nan         Rape   

                  timestamp x_coordinate y_coordinate  year zipcode  
0 2016-01-19 12:00:00+00:00         None         None  2016    None  
1 2016-01-25 12:00:00+00:00         None         None  2016    None  
2 2016-01-25 12:00:00+00:00         None         None  2016    None  
3 2016-01-27 12:00:00+00:00         None         None  2016    None  
4 2016-02-01 12:00:00+00:00         None         None  2016    None

### 2. Apa saja jenis tindak kejahatan yang terjadi di kota Austin selama tahun 2016?

In [4]:
df2 = client.query('''
SELECT DISTINCT description, year
FROM `bigquery-public-data.austin_crime.crime` 
WHERE year = 2016
GROUP BY year , description
''').to_dataframe()

df2

description  year
0   RAPE                            2016
1   AGG RAPE OF A CHILD             2016
2   RAPE OF A CHILD                 2016
3   SEXUAL ASSAULT W/ OBJECT        2016
4   SEXUAL ASSAULT OF CHILD/OBJECT  2016
5   AGG RAPE                        2016
6   AGG SEXUAL ASSAULT CHILD/OBJEC  2016
7   THEFT FROM PERSON               2016
8   AGG ROBBERY/DEADLY WEAPON       2016
9   THEFT                           2016
10  AGG ASLT STRANGLE/SUFFOCATE     2016
11  BURGLARY OF VEHICLE             2016
12  BURGLARY NON RESIDENCE          2016
13  THEFT OF METAL                  2016
14  ROBBERY BY ASSAULT              2016
15  THEFT FROM AUTO                 2016
16  AUTO THEFT                      2016
17  THEFT BY SHOPLIFTING            2016
18  THEFT OF BICYCLE                2016
19  THEFT OF TRAILER                2016
20  THEFT OF LICENSE PLATE          2016
21  BURGLARY OF RESIDENCE           2016
22  THEFT OF HEAVY EQUIPMENT        2016
23  THEFT OF AUTO PARTS             2016
24  BURG NON RESIDENCE SHEDS        2016
25  AGG ASSAULT FAM/DATE VIOLENCE   2016
26  ROBBERY BY THREAT               2016
27  BREACH OF COMPUTER SECURITY     2016
28  AGG ASSAULT                     2016
29  AGG ASSAULT ON PUBLIC SERVANT   2016
30  MANSLAUGHTER                    2016
31  FORCED SODOMY                   2016
32  THEFT FROM BUILDING             2016
33  DEADLY CONDUCT                  2016
34  AGG ASSAULT WITH MOTOR VEH      2016
35  POCKET PICKING                  2016
36  PURSE SNATCHING                 2016
37  AGG ASLT ENHANC STRANGL/SUFFOC  2016
38  BURG OF RES - SEXUAL NATURE     2016
39  MURDER                          2016
40  AGG ROBBERY BY ASSAULT          2016
41  AGG ASLT W/MOTOR VEH FAM/DAT V  2016
42  BURGLARY OF COIN-OP MACHINE     2016
43  MISAPPLY FIDUCIARY PROP         2016
44  AGG SEXUAL ASSAULT W OBJECT     2016
45  THEFT CATALYTIC CONVERTER       2016
46  TAKE WEAPON FRM POLICE OFFICER  2016
47  THEFT/TILL TAPPING              2016
48  CAPITAL MURDER                  2016
49  FORCED SODOMY OF CHILD          2016
50  AGG FORCED SODOMY               2016
51  DEADLY CONDUCT FAM/DATE VIOL    2016
52  AGG FORCED SODOMY OF CHILD      2016
53  SERIOUS INJURY TO A CHILD       2016

> Berdasarkan hasil output diatas di peroleh informasi bahwa jenis tindak kejahatan yang terjadi di kota Austin selama tahun 2016 sebanyak 54 jenis kejahatan. Hal ini dapat menjadikan pertimbangan oleh pihak kepolisian

### 3. Berapa banyak kasus tindak kejahatan berdasarkan kasus pembersihannya (clearance status) selama tahun 2016?

In [5]:
df3 = client.query('''
SELECT 	COUNT(clearance_status) AS kasus_tindak_kejahatan
FROM `bigquery-public-data.austin_crime.crime` 
WHERE year = 2016
''').to_dataframe()

df3

kasus_tindak_kejahatan
0                   35742

> Berdasarkan hasil output diatas di peroleh informasi bahwa terdapat kasus tindak kejahatan berdasarkan kasus pembersihannya 35742 kasus, hal ini dapat menjadikan bahan pertimbangan pihak kepolisian guna memberikan treatment yang lebih supaya kasus tindakan kejahatan berkurang

### 4. Jenis kejahatan apa saja yang paling banyak belum terselesaikan kasusnya di tahun 2016?

In [7]:
df4 = client.query('''
SELECT year, description,
COUNT(clearance_status) AS total_belum_terselesaikan
FROM `bigquery-public-data.austin_crime.crime` 
WHERE year = 2016 AND clearance_status = 'Not cleared'
GROUP BY year, description
ORDER BY total_belum_terselesaikan DESC
''').to_dataframe()

df4

year                     description  total_belum_terselesaikan
0   2016  THEFT                                                8607
1   2016  BURGLARY OF VEHICLE                                  8089
2   2016  BURGLARY OF RESIDENCE                                2706
3   2016  THEFT BY SHOPLIFTING                                 1565
4   2016  AUTO THEFT                                           1540
5   2016  BURGLARY NON RESIDENCE                               1441
6   2016  THEFT OF BICYCLE                                     1346
7   2016  THEFT FROM AUTO                                       633
8   2016  AGG ASSAULT                                           530
9   2016  THEFT OF LICENSE PLATE                                445
10  2016  THEFT FROM PERSON                                     362
11  2016  AGG ROBBERY/DEADLY WEAPON                             298
12  2016  BURG NON RESIDENCE SHEDS                              287
13  2016  THEFT OF AUTO PARTS                                   283
14  2016  THEFT FROM BUILDING                                   241
15  2016  ROBBERY BY ASSAULT                                    210
16  2016  THEFT OF TRAILER                                      151
17  2016  AGG ASLT STRANGLE/SUFFOCATE                           144
18  2016  RAPE                                                  128
19  2016  DEADLY CONDUCT                                        114
20  2016  BREACH OF COMPUTER SECURITY                            87
21  2016  AGG ASSAULT FAM/DATE VIOLENCE                          70
22  2016  SEXUAL ASSAULT W/ OBJECT                               64
23  2016  SEXUAL ASSAULT OF CHILD/OBJECT                         57
24  2016  RAPE OF A CHILD                                        52
25  2016  ROBBERY BY THREAT                                      44
26  2016  AGG ASSAULT WITH MOTOR VEH                             39
27  2016  THEFT OF METAL                                         36
28  2016  BURGLARY OF COIN-OP MACHINE                            25
29  2016  FORCED SODOMY                                          20
30  2016  AGG RAPE OF A CHILD                                    12
31  2016  AGG SEXUAL ASSAULT CHILD/OBJEC                         12
32  2016  THEFT OF HEAVY EQUIPMENT                               11
33  2016  AGG ASLT W/MOTOR VEH FAM/DAT V                          8
34  2016  AGG ROBBERY BY ASSAULT                                  8
35  2016  AGG RAPE                                                7
36  2016  AGG ASLT ENHANC STRANGL/SUFFOC                          7
37  2016  BURG OF RES - SEXUAL NATURE                             6
38  2016  AGG ASSAULT ON PUBLIC SERVANT                           6
39  2016  AGG SEXUAL ASSAULT W OBJECT                             5
40  2016  PURSE SNATCHING                                         4
41  2016  POCKET PICKING                                          3
42  2016  MISAPPLY FIDUCIARY PROP                                 3
43  2016  THEFT/TILL TAPPING                                      2
44  2016  AGG FORCED SODOMY                                       2
45  2016  THEFT CATALYTIC CONVERTER                               1
46  2016  DEADLY CONDUCT FAM/DATE VIOL                            1
47  2016  AGG FORCED SODOMY OF CHILD                              1
48  2016  MURDER                                                  1
49  2016  FORCED SODOMY OF CHILD                                  1

> Berdasarkan hasil output diatas di peroleh informasi bahwa jenis kejahatan yang paling banyak belum terselesaikan kasusnya di tahun 2016 adalah jenis kejahatan **THEFT** atau pencurian yaitu sebanyak 8607. Hal ini perlu diperhatikan oleh pihak kepolisisan guna segera dapat terselesaikan

### 5. Di bulan apa di tahun 2016 kasus pencurian (theft) sangat banyak terjadi?

In [12]:
df5 = client.query('''
SELECT  year, EXTRACT(month FROM timestamp) AS month, description,
COUNT(description) AS total_kasus_pencurian 
FROM `bigquery-public-data.austin_crime.crime` 
WHERE year = 2016 AND description = 'THEFT                         '
GROUP BY year,month, description
ORDER BY total_kasus_pencurian DESC
''').to_dataframe()

df5

year  month                     description  total_kasus_pencurian
0   2016     12  THEFT                                             925
1   2016      4  THEFT                                             900
2   2016      6  THEFT                                             898
3   2016      3  THEFT                                             884
4   2016     10  THEFT                                             882
5   2016     11  THEFT                                             881
6   2016      9  THEFT                                             874
7   2016      1  THEFT                                             872
8   2016      7  THEFT                                             864
9   2016      8  THEFT                                             858
10  2016      5  THEFT                                             852
11  2016      2  THEFT                                             752

> Berdasarkan hasil output diatas di peroleh informasi bahwa kasus pencurian (theft) yang sangat banyak terjadi di tahun 2016 adalah pada bulan Desember yaitu sebanyak 925. Hal ini bisa saja disebabkan meningkatnya pengunjung di kota Austin yang bertepatan dengan peringataan hari natal atau cuti bersama, sehingga kasus pencurian di kota Austin meningkat. Perihal ini  perlu menjadi perhatian khusus bagi pihak kepolisisan untuk meningkatkan keamaan dibulan - bulan tertentu, guna meminimalisir tinggak kejahatan


In [16]:
df5_1 = client.query('''
SELECT SUM(total_kasus_pencurian) AS jumlah_kasus_pencurian
FROM(SELECT  year, EXTRACT(month FROM timestamp) AS month, description,
COUNT(description) AS total_kasus_pencurian 
FROM `bigquery-public-data.austin_crime.crime` 
WHERE year = 2016 AND description = 'THEFT                         '
GROUP BY year,month, description
ORDER BY total_kasus_pencurian DESC)
''').to_dataframe()

df5_1

jumlah_kasus_pencurian
0                   10442

> Berdasarkan hasil output diatas di peroleh informasi bahwa jumlah total kasus pencurian pada tahun 2016 adalah sebanyak 10442 kasus

### 6. Di distrik apa yang paling banyak terjadi kasus pencurian selama 2016?

In [13]:
df6 = client.query('''
SELECT year, district,
COUNT(description) AS total_kasus_pencurian
FROM `bigquery-public-data.austin_crime.crime` 
WHERE year = 2016 AND description = 'THEFT                         '
GROUP BY year, district
ORDER BY total_kasus_pencurian DESC
''').to_dataframe()

df6

year district  total_kasus_pencurian
0   2016        D                   1597
1   2016        F                   1230
2   2016        I                   1140
3   2016        B                   1131
4   2016        A                   1121
5   2016        G                   1121
6   2016        E                   1113
7   2016        H                   1050
8   2016        C                    778
9   2016       AP                    117
10  2016       UK                     36
11  2016       88                      8

> Berdasarkan hasil output diatas di peroleh informasi bahwa kasus pencurian terbanyak yang terjadi pada tahun 2016 terjadi di distrik D sebanyak 1597 kasus pencurian. Hal ini perlu di perhatikan oleh pihak kepolisian guna meningkatkan keamaan di distrik D, sehingga terjadi penurunan kasus pencurian atau jenis kejahatan lainnya

### 7. Berapa lama rata-rata waktu (dalam hari) kasus tindak kejahatan 'theft' terselesaikan sejak kasus tersebut terangkat selama tahun 2016? (Hint: Hitung selisih hari dari timestamp hingga clearance date)

In [14]:
df6 = client.query('''
SELECT AVG(selisih_hari) AS rata_rata_selisih_hari
FROM(SELECT year, description, timestamp, clearance_date,
DATE_DIFF(clearance_date, timestamp, day) AS selisih_hari
FROM `bigquery-public-data.austin_crime.crime` 
WHERE year = 2016 AND description = 'THEFT                         ')
''').to_dataframe()

df6

rata_rata_selisih_hari
0               21.337341

> Berdasarkan hasil output diatas di peroleh informasi bahwa lama rata-rata waktu (dalam hari) kasus tindak kejahatan 'theft' terselesaikan sejak kasus tersebut terangkat selama tahun 2016 adalah 21 hari, yang berarti pihak kepolisian dapat menyelesaikan kasus kejahatan jenis pencurian selama rata-rata 21 hari pada tahun 2016

# **Conclusions**

> Berdasarkan hasil analisis di atas dapat disimpulkan bahwa, jenis kejahatan yang terjadi di kota Austin pada tahun 2016 ada sebanyak 54 jenis kejahatan, dan yang paling banyak kasus kejahatan yang belum terselesaikan adalah jenis kejahatan pencurian, yaitu sebanyak 8607 kasus. Sedangkan total jumlah kasus pencurian di kota Austin adalah sebanyak 10442 kasus.

> Dari hasil analisis di atas juga diketahui bahwa distrik yang paling banyak terjadi kasus pencurian selama 2016 adalah distrik D sebanyak 1597 kasus pencurian. Dan rata - rata pihak kepolisian dapat menyelesaikan kasus kejahatan jenis pencurian di kota Austin selama 21 hari 